File: video-service-client/src/main/java/com/boclips/videos/service/client/CreateCollectionRequest.java
package com.boclips.videos.service.client;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.util.List;
import java.util.Set;

@Builder
@Data
public class CreateCollectionRequest {
    private final @NonNull String title;
    private final String description;
    private final List<String> videos;
    private final Set<String> subjects;
    private final boolean isPublic;
}

File: video-service/src/main/kotlin/com/boclips/videos/service/presentation/collections/CreateCollectionRequest.kt
package com.boclips.videos.service.presentation.collections

import javax.validation.constraints.NotBlank

data class CreateCollectionRequest(
    @field:NotBlank(message = "Title is required")
    val title: String?,
    val description: String? = null,
    val videos: List<String> = emptyList(),
    val public: Boolean? = null,
    val subjects: Set<String> = emptySet()
)
